### ISSUE TITLE
Timelines - Tween containers

### ISSUE THREAD

**Issue by b005t3r** [Gamua/Starling-Framework#914]
Chaining multiple tweens, so each one starts after the previous one finished is currently quite fiddly - you basically have to start a new tween from the old one's onComplete handler - not very convenient. It's also currently impossible to create a more complex animations/sequences and play them back and forth, change their transition, etc.

My proposition is to introduce a container class for storing tweens (and other containers) - a Timeline class. They have something like that in this Java lib: http://www.aurelienribon.com/blog/projects/universal-tween-engine/

It's really a must have if you plan to do any more complex animations, so I think this would make a great addition to Starling's tweens.

**Comment by pSi-X** [comment]
Look at https://github.com/mayakwd/as3-eaze-tween

**Comment by b005t3r** [comment]
Yeah, but the problem is this is not a part of Starling and Starling's tweening engine is already pretty solid, it just needs timelines.

**Comment by PrimaryFeather** [comment]
Thanks for the suggestion! I can totally see the sense in that.
Anybody who wants do see this, please vote!

BTW, just for reference: it's by far not as elaborate, but there actually is a way you can chain tweens together in the current API: via [Tween.nextTween](http://doc.starling-framework.org/current/starling/animation/Tween.html#nextTween).

**Comment by b005t3r** [comment]
```Tween.nextTween()``` is OK, but it's not as functional as timelines. It's, for instance, much more inconvenient to pause a complex animation created with nextTweens than to pause a timeline-based one.

If you need any more references, I created something like that some time ago (based on your Tweens), but it's still inconvenient to use, if it's not a part of Starling:

https://github.com/b005t3r/Stork-Extension-Tween/blob/master/extension/src/stork/tween/TimelineNode.as

**Comment by PrimaryFeather** [comment]
Thanks, I will definitely take a look at that! :smile:

**Comment by denisgl777** [comment]
vote!

**Comment by denisgl777** [comment]
Daniel
Do you consider adding this function in the future?

**Comment by PrimaryFeather** [comment]
I'm definitely considering that, yes! I can totally see the use of it. As for the priority — anyone, please just keep voting! :smile:

**Comment by naderalbes** [comment]
vote vote

**Comment by DQvsRA** [comment]
Any update on Tween Timeline, executing multiple tweens in a sequence?

**Comment by PrimaryFeather** [comment]
Not yet, sorry! But I'll take this as a "vote"! :wink:

**Comment by jimmymjing** [comment]
here's my vote! Currently I just have to use TimelineMax...

**Comment by Shaunmax** [comment]
Vote Vote Vote!

**Comment by Mintonist** [comment]
Vote!)

**Comment by chachawu** [comment]
vote!

**Comment by Adolio** [comment]
+1

**Comment by denisgl7** [comment]
Vote
